syntax = "proto3";

message Request{
    int32 client_id = 1;
    string req_msg = 2;
}

message commonResponse{
    bool success = 1;
    string res_msg = 2;
}

message scoreInfo{
    int32 rgroup_id = 1;
    int32 rank_id = 2;
    float rank_score = 3;
    repeated float raw_score = 4;
    repeated float vol_score = 5;
}

message volumeInfo{
    string folder_name = 1;
    string folder_path = 2;
    repeated int32 dims = 3;/*height,width,depth*/
    repeated float orientation = 4;
    repeated float resolution = 5;/*pix_spacing y, x, slice thickness*/
    float volume_loc_range = 6;
    bool with_mask = 7;
    enum DataSource{
        LOCAL = 0;
        SERVER = 1;
        DEVICE = 2;
    }
    DataSource data_source = 8;
    bytes sample_img = 9;
    scoreInfo scores = 10;
}